### PR TITLE
warning when the parameter could not be converted

### PR DESCRIPTION
**Pull request by MichaelPFrey**
bildschirmfoto von 2018-11-16 21-45-17

this partially addresses #2575

### CODE REVIEW

File: src/primitives.cc [hunk 1]
```diff
@@ -190,10 +190,16 @@ AbstractNode *PrimitiveModule::instantiate(const Context *ctx, const ModuleInsta
...
-		size->getDouble(node->z);
-		size->getVec3(node->x, node->y, node->z);
+		if(size != ValuePtr::undefined){
+			bool converted=false;
+			converted |= size->getDouble(node->x);
+			converted |= size->getDouble(node->y);
+			converted |= size->getDouble(node->z);
+			converted |= size->getVec3(node->x, node->y, node->z);
+			if(!converted){
+				PRINTB("WARNING: Unable to convert CUBE size=%s to a number or a vec3 of numbers, %s", size->toString() % inst->location().toString());
```
Comment (kintel) [inline src/primitives.cc]: I wonder if we should print `CUBE` in lower case here, perhaps even `cube(size=%s)`.
Same goes for other similar print statements.

**Comment by kintel** [comment]
@nophead You tend to have valuable input on console output. Any concerns with this?

File: src/primitives.cc (hunk 1 above)
Comment (MichaelPFrey) [reply src/primitives.cc]: `cube` is fine by me. `cube(size=%s)` can get confusing, as we do not show the state of for e.g. `center`. Note that cube is the simplest case, other modules have a much more complicated signature.
We could use [Ellipsis](https://en.wikipedia.org/wiki/Ellipsis) to hint add the omission like so `cube(size=%s...)` or so `cube(size=%s, ...)`. Note that we have also have highly complicated built ins like:
`polyhedron(..., triangles = [[0,1,2,3],[5,4,3,2],[0,4,5,1],[0,3,4],[5,2]], ...);`.

Note that it gets ugly when strings come in to play (`cube("[1,2,3]");`) - but this is an issue either-way.

**Comment by MichaelPFrey** [comment]
A screenshot showing why the line number is not always enough to deduce the issue: (partially #2392)
bildschirmfoto von 2018-11-19 21-23-18
```
l = 61;
w = 44;
h = 78 + 7;

thick = 1;

difference() {
    cube([l+thick*2, w+thick*2, h+thick]);
    translate([thick, thick, thick])
       cube([l, w. h+thick*2]);
}

array = [1,[1],[1,2],[1,2,3]];

for (i = [1:4]) {
    cube(array[i]);
}
```

**Comment by MichaelPFrey** [comment]
bildschirmfoto von 2018-11-19 22-34-39

**Pull request merged**